dspace-rest/src/main/java/org/dspace/rest/ItemsResource.java

```diff
@@ -992,7 +992,7 @@ else if (!AuthorizeManager.authorizeActionBoolean(context, bitstream, org.dspace
             {
                 sql += "QUALIFIER='" + metadata[2] + "' AND ";
             }
-            sql += "dbms_lob.substr(TEXT_VALUE, 40)='" + metadataEntry.getValue() + "' AND ";
+            sql += "TEXT_VALUE='" + metadataEntry.getValue() + "' AND ";
```

Comment:
I agree. I tried the oracle string conversion functions for VARCHAR2, but without success. Always ended with an error ORA-00932. I think that is no way to use identical code for both DBMS, except change datatype CLOB to VARCHAR2 of column TEXT_VALUE of table METADATAVALUE.

I think that better solution for oracle will be:
```sql
sql += "dbms_lob.compare(TEXT_VALUE, '" + metadataEntry.getValue() + "') = 0 AND ";
```
or maybe
```sql
sql += "dbms_lob.compare(TEXT_VALUE, to_clob('" + metadataEntry.getValue() + "')) = 0 AND ";
```